src/core/shape_inference/include/stft_shape_inference.hpp

```diff
@@ -0,0 +1,99 @@
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           frame_size_shape.rank().compatible(0),
+                           "The shape of frame_size must be a scalar.");
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           frame_step_shape.rank().compatible(0),
+                           "The shape of frame_step must be a scalar.");
+
+    const auto frame_size = get_input_const_data_as<TRShape, int64_t>(op, 2, ta);
+    const auto frame_step = get_input_const_data_as<TRShape, int64_t>(op, 3, ta);
+
+    if (signal_shape.rank().is_dynamic()) {
+        return {signal_shape};
+    } else if (!frame_size || !frame_step) {
+        return {TRShape{signal_shape[0], -1, -1, 2}};
+    }
+
+    const auto& frame_size_val = (*frame_size)[0];
+    const auto& frame_step_val = (*frame_step)[0];
+
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           0 < frame_size_val && (signal_shape.rank().is_dynamic() ||
+                                                  frame_size_val < signal_shape[1].get_interval().get_max_val()),
+                           "Provided frame size is ",
+                           frame_size_val,
+                           " but must be in range [1, ",
+                           signal_shape[1],
+                           "].");
+
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           0 < frame_step_val,
+                           "Provided frame step is ",
+                           frame_step_val,
+                           " but must be greater than zero.");
+
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           window_shape.is_dynamic() ||
+                               (0 < window_shape[0].get_length() && window_shape[0].get_length() <= frame_size_val),
+                           "Window input dimension must be in range [1, ",
+                           frame_size_val,
+                           "].");
+
+    const auto& batch_dim = signal_shape[0];
+    const TDim frame_size_dim = TDim{frame_size_val};
+    const TDim signal_frame_size_diff = signal_shape[1] - frame_size_dim;
+    const TDim fft_samples_dim = (frame_size_val / 2) + 1;
+
+    // Divsion opeartor for static Dimension of PartialShape can return non static dimension and ceil instead of floor
+    // for lower bound, so floor_div util is used instead
+    const TDim frames_dim = ov::util::dim::floor_div(signal_frame_size_diff, frame_step_val) + 1;
+
+    TRShape output_shape;
+    if (op->get_transpose_frames()) {
+        output_shape = TRShape{std::move(batch_dim), std::move(fft_samples_dim), std::move(frames_dim), 2};
+    } else {
+        output_shape = TRShape{std::move(batch_dim), std::move(frames_dim), std::move(fft_samples_dim), 2};
+    }
+    return {std::move(output_shape)};
```

Comment:
Detail: Why not created vector at beginning then emplace required shape and not use move at all?

Reply:
Discussed offline and applied